mne_bids/tests/test_write.py

```diff
@@ -123,13 +123,44 @@ def test_make_dataset_description():
                                          'or "derivative."'):
         make_dataset_description(path=tmp_dir, name='tst', dataset_type='src')
 
+    make_dataset_description(path=tmp_dir, name='tst')
+
+    with open(op.join(tmp_dir, 'dataset_description.json'), 'r') as fid:
+        dataset_description_json = json.load(fid)
+        assert dataset_description_json["Authors"] == \
+            ["Please cite MNE-BIDS in your publication before removing this "
+             "(citations in README)"]
+
     make_dataset_description(
         path=tmp_dir, name='tst', authors='MNE B., MNE P.',
         funding='GSOC2019, GSOC2021',
         references_and_links='https://doi.org/10.21105/joss.01896',
-        dataset_type='derivative', verbose=True
+        dataset_type='derivative', overwrite=False, verbose=True
+    )
+
+    with open(op.join(tmp_dir, 'dataset_description.json'), 'r') as fid:
+        dataset_description_json = json.load(fid)
+        assert dataset_description_json["Authors"] == \
+            ["Please cite MNE-BIDS in your publication before removing this "
+             "(citations in README)"]
+
+    make_dataset_description(
+        path=tmp_dir, name='tst2', authors='MNE B., MNE P.',
+        funding='GSOC2019, GSOC2021',
+        references_and_links='https://doi.org/10.21105/joss.01896',
+        dataset_type='derivative', overwrite=True, verbose=True
     )
 
+    with open(op.join(tmp_dir, 'dataset_description.json'), 'r') as fid:
+        dataset_description_json = json.load(fid)
+        assert dataset_description_json["Authors"] == ['MNE B.', 'MNE P.']
+
+    with pytest.raises(ValueError, match='Previous BIDS version used'):
+        version = make_dataset_description.__globals__['BIDS_VERSION']
+        make_dataset_description.__globals__['BIDS_VERSION'] = 'old'
```

Comment:
why this?

Reply:
To put it back in case it would cause issues for later tests

Reply:
Maybe a reimport would be better? Importlib hasn't been used yet though so maybe best to limit dependencies if possible

Reply:
The tests pass without it though too, I can take it out I just don't want unexpected behavior down the road

Reply:
oh I get it now. You can keep it. Wouldn't the coverage reduce without it?

Reply:
maybe add a comment to explain that you need to do this because the variable BIDS_VERSION has already been set during import

Reply:
Without it will reduce coverage but as long as there are no tests after it that use make_dataset_description there shouldn't be any errors... we can revert the last commit